### PR TITLE
Implement page.darkBackground and page.overlayMenu with CSS and data attributes

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Migrate menuOverlay and darkBackground page flags to DOM attributes, similar to previous PR

Fix footer bg color for dark background page

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4122?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4123** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4123?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4122** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4122?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4121** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4121?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4120** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4120?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/app/[locale]/StoryblokLayout.css.ts [hunk 1]
```diff
@@ -1,8 +1,25 @@
 import { style } from '@vanilla-extract/css'
-
-export { dark } from 'ui/src/theme/dark.css'
```
Comment (alebedev) [inline apps/store/src/app/[locale]/StoryblokLayout.css.ts:3]: It's easier to copy-paste this block between pages and app routes vs keeping it as dependency.  Right now separate style became a nested selector and those are harder to shart

File: apps/store/src/blocks/PageBlock.tsx [hunk 2]
```diff
@@ -14,6 +14,9 @@ export const PageBlock = ({ blok }: PageBlockProps) => {
         className={main}
         {...storyblokEditable(blok)}
         data-hide-breadcrumbs={!!blok.hideBreadcrumbs}
+        data-dark-background={!!blok.darkBackground}
+        data-overlay-menu={!!blok.overlayMenu}
+        data-hide-chat={!!blok.hideChat}
```
Comment (alebedev) [inline apps/store/src/blocks/PageBlock.tsx:19]: Not used yet, but we'll need it later

File: apps/store/src/components/Header/Header.css.ts [hunk 3]
```diff
@@ -1,5 +1,4 @@
 import { createVar, style } from '@vanilla-extract/css'
-import { recipe } from '@vanilla-extract/recipes'
```
Comment (alebedev) [inline apps/store/src/components/Header/Header.css.ts:2]: Keeping '@vanilla-extract/recipes' in dependencies for now, we should remove it if we don't use it in the coming weeks

File: apps/store/src/components/LayoutWithMenu/Breadcumbs.css.ts [hunk 4]
```diff
@@ -15,7 +15,7 @@ export const breadcrumbsList = style({
       content: '""',
       margin: 'auto',
     },
-    [`main[data-hide-breadcrumbs=true] ~ &`]: {
+    [`body:has(main[data-hide-breadcrumbs=true]) &`]: {
```
Comment (alebedev) [inline apps/store/src/components/LayoutWithMenu/Breadcumbs.css.ts:18]: Unified with selector in Header.  Sibling selector only works when target block is located after sibling - OK for footer, but not for header

**Review by gustaveen — APPROVED** [review]
Nice!

**Comment by alebedev** [comment]
### Merge activity

* **Apr 5, 7:22 AM EDT**: @alebedev started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4122).

**Review by guilhermespopolin — APPROVED** [review]

**Pull request merged**